### DISCUSSION TITLE
Performance style

### DISCUSSION THREAD

**Discussion by infojunkie** [w3c-cg/musicxml#449] [Ideas]
(This is a paraphrase of a [mailing list post](https://lists.w3.org/Archives/Public/public-music-notation/2020Nov/0002.html), to which I was invited to post an issue here).

I am working on Web-based rendering and automated playback of lead sheets, and I'm using MusicXML as the main interchange format between various apps (for notation, conversion, display, playback). 

A typical lead sheet contains hints regarding the style of performance, e.g. Ballad, Bossa Nova, Funk, etc. Also typically, playback engines such as iReal Pro, Band-in-a-Box and others are able to convert this information to a multi-track accompaniment to the notated music.

MusicXML 3.1 does not include a metadata element to convey the performance style. Anecdotally, iReal Pro converts their sheet style to a "creator lyricist" element, to ensure it displays in the expected place when imported. 

But really, what's needed is an explicit element type that can be machine-read. Ideally, it would be an element that attaches to a timed event, to allow the style to change over the course of the performance.

**Comment by infojunkie** [comment]
For reference, iReal Pro provides the following built-in styles, with the possibility of adding custom ones:
Afoxe, Afro, Baião, Ballad, Bossa Nova, Chacarera, Even 8ths, Funk, Latin, Medium Swing, Medium Up Swing, Pop, Pop Ballad, Reggae, RnB, Rock, Rock Pop, Samba, Samba Funk, Shuffle, Slow Bossa, Slow Swing, Up Tempo Swing, Waltz.

The [open source playback engine Musical MIDI Accompaniment (MMA)](https://www.mellowood.ca/mma/) defines the performance style as the "Groove" that translates into track specifications for drums, bass and piano. The MMA groove applies to chord progressions and can be [programmed using a scripting language](https://www.mellowood.ca/mma/online-docs/html/tut/node5.html#SECTION00510000000000000000). MMA includes a large number of common "grooves".

**Comment by infojunkie** [comment]
In typical lead sheets, such as Sher's New Real Book series, the performance style is typically annotated next to the tempo specification, and usually with the same visual style. Maybe `performance-style (#PCDATA)` could be a child of `metronome` ?

**Comment by mogenslundholm** [comment]
There seems to be some elements with information not known by MusicXML.
They have names like other-play, other-appearance, other-notation,
other-articulation etc..
The next version of MusicXML will be 4.0. I think this version might
have some of the wanted functionality.

**Comment by infojunkie** [comment]
More information here: I discovered that iReal Pro has 2 notions of performance style: [one for display (which I copied in a previous comment)](https://github.com/infojunkie/ireal-musicxml/blob/main/doc/ireal.md#styles-for-display), and [one that specifies the actual playback "groove"](https://github.com/infojunkie/ireal-musicxml/blob/main/doc/ireal.md#grooves-for-playback).

**Comment by infojunkie** [comment]
As I've started working on a [MusicXML to MIDI accompaniment generator](https://github.com/infojunkie/musicxml-mma), I am back thinking about this issue. Some leadsheets (e.g. Bolivia) specify several performance styles at different sections. To remain consistent with other elements, the performance element should probably exist both within `direction-type` and `sound`. 

![Bolivia](https://user-images.githubusercontent.com/92977/103036201-0c24b600-451e-11eb-9a15-486c20dbef70.png)

**Comment by mdgood** [comment]
@infojunkie Thank you for raising this issue. I think that classifying performance styles in a way that works well across applications could be tricky. Each application already has its own way of doing things, and the commonalities can be difficult to standardize. An alternative would be a free text field, but I suspect that would not do what you want. 

I don't think we have time to do this for MusicXML 4.0. It is an interesting idea for a later version and/or for MNX.

In the meantime, I think you can use the `<other-play>` element to encode this in your application, as @mogenslundholm suggested above.

**Comment by infojunkie** [comment]
Thanks. I look forward to discussing "grooves" in MusicXML / MNX !

**Comment by infojunkie** [comment]
Could we please convert this question to a forum discussion ? :pray:

**Comment by mdgood** [comment]
Sure - done.

**Comment by infojunkie** [comment]
Thanks for your suggestions here. I've updated my [iReal Pro to MusicXML converter](https://github.com/infojunkie/ireal-musicxml) to emit `direction/direction-type/words` for visual indication of performance style changes, and `sound/play/other-play[@type = 'groove']` to indicate playback style changes.  Also, my [MusicXML to Musical MIDI Accompaniment (MMA) converter](https://github.com/infojunkie/musicxml-mma) now parses `sound/play/other-play[@type = 'groove']` to emit playback style (aka groove) declarations.
